File: Homeworks/Homework1/Tire_Kutay.lua
-- Function declaration of foo()

function foo(key, value)
  print(key, ":", value)
end


-- Initialization of Associative Array

luaDictionary = {}
luaDictionary["Assyria"] = "Nineveh"
luaDictionary["Babylonia"] = "Babylon"
luaDictionary["Sumeria"] = "Ur"
luaDictionary["Persia"] = "Susa"

-- Getting a value

print("\nGetting the value of 'Sumeria'")
local value1 = luaDictionary["Sumeria"]
print("The value of Sumeria is:", value1)

-- Adding a new element

print("\nAdding the new element 'Egypt': ")
luaDictionary['Egypt'] = "Thebes"
for key, value in pairs(luaDictionary) do
       foo(key,value)
end

-- Removing an element

print("\nRemoving the element 'Persia': ")
luaDictionary['Persia'] = nil
for key, value in pairs(luaDictionary) do
       foo(key,value)
end

-- Modifying an element

print("\nModifying the value of 'Assyria': ")
luaDictionary['Assyria'] = "Assur"
for key, value in pairs(luaDictionary) do
       foo(key,value)
end

-- Searching for the existence of a key

print("\nSearching for the 'Babylonia' key:")
local foundKey = false

for key,value in pairs (luaDictionary) do
 if key == 'Babylonia' then
   foundKey = true
   break
 end
end

if foundKey then
   print("The key is successfully found.\n")
else
 print("Key was not found.\n")
end

-- The key is not found

print("\nSearching for the 'Rome' key:")
local foundKey2 = false

for key,value in pairs (luaDictionary) do
 if key == 'Rome' then
   foundKey2 = true
   break
 end
end

if foundKey2 then
   print("The key is successfully found.\n")
else
 print("Key was not found.\n")
end
     
-- Searching for the existence of a value

print("\nSearching for the 'Thebes' value:")
local foundValue = false

for key,value in pairs (luaDictionary) do
 if value == 'Thebes' then
   foundValue = true
   break
 end
end

if foundValue then
   print("The value is successfully found.\n")
else
 print("Value was not found.\n")
end

-- The value is not found

print("\nSearching for the 'Athens' value:")
local foundValue2 = false

for key,value in pairs (luaDictionary) do
 if value == 'Athens' then
   foundValue2 = true
   break
 end
end

if foundValue2 then
   print("The value is successfully found.\n")
else
 print("Value was not found.\n")
end


-- printing key-value pair

print("Looping through the array: ")
for key, value in pairs(luaDictionary) do
       foo(key,value)
end

File: Homeworks/Homework1/Tire_Kutay.py
# Associative Array Initialization

pythonDictionary = {
    
    'Assyria' : 'Nineveh',
    'Babylonia' : 'Babylon',
    'Sumeria' : 'Ur',
    'Persia' : 'Susa'
}
  
# Getting a value

print("Getting the value of 'Sumeria'")
val1 = pythonDictionary['Sumeria']
print("The value of 'Sumeria' is: ", val1)


print("\nGetting the value of 'Persia'")
val2 = pythonDictionary.get('Persia')
print("The value of 'Persia' is: ", val2)

# Adding a new element

print("\nAdding a new element 'Egypt': ")
pythonDictionary['Egypt'] = "Thebes"
print(pythonDictionary)

# Removing an element

print("\nRemoving the element 'Persia': ")
removedElement = pythonDictionary.pop('Persia')
print(pythonDictionary)

# Modifying an element

print("\nModifying the element 'Assyria': ")
pythonDictionary['Assyria'] = "Assur"
print(pythonDictionary)

# Searching for the existence of a key

print("\nSearching for 'Babylonia' key: ")
if "Babylonia" in pythonDictionary:
    print("'Babylonia' key is found with the value: ", pythonDictionary['Babylonia'])
else:
    print("'Babylonia' key was not found.")
    
# The key is not found

print("\nSearching for 'Rome' key: ")
if "Rome" in pythonDictionary:
    print("'Rome' key is found with the value: ", pythonDictionary['Rome'])
else:
    print("'Rome' key was not found.")

# Searching for the existence of a value
print("\nSearching for 'Thebes' value: ")
if "Thebes" in pythonDictionary.values():
    print("'Thebes' value is found.")
else:
    print("'Thebes' was not found as a value.")
    
# The value is not found

print("\nSearching for 'Athens' value: ")
if "Athens" in pythonDictionary.values():
    print("'Athens' value is found.")
else:
    print("'Athens' was not found as a value.")

# Function declaration of foo() and printing key-value 

print("\nLooping through the array: ")
def foo(key, value):
    print(key, ": ", value)

for key in pythonDictionary:
    foo(key, pythonDictionary[key])

